lib/dotcom_web/components/trip_planner/itinerary_detail.ex

```diff
@@ -77,15 +78,19 @@ defmodule DotcomWeb.Components.TripPlanner.ItineraryDetail do
         <.route_symbol :for={route <- @all_routes} route={route} class="ml-2" />
       </div>
       <div :for={leg <- @itinerary.legs}>
-        <.leg
-          start_time={leg.start}
-          end_time={leg.stop}
-          from={leg.from}
-          to={leg.to}
-          mode={leg.mode}
-          realtime={leg.realtime}
-          realtime_state={leg.realtime_state}
-        />
+        <%= if match?(%PersonalDetail{}, leg.mode) do %>
+          <.walking_leg leg={leg} />
+        <% else %>
+          <.leg
+            start_time={leg.start}
+            end_time={leg.stop}
+            from={leg.from}
+            to={leg.to}
+            mode={leg.mode}
+            realtime={leg.realtime}
+            realtime_state={leg.realtime_state}
+          />
```

Comment:
📓 We should probably rename this to TransitLeg and not destructure the leg.

Reply:
did you look inside `DotcomWeb.Components.TripPlanner.Leg.leg/1`? it's basically a wrapper that shows a transit leg or walking leg. Might want to either (a) move this to there or (b) simplify `leg/1` internal logic to remove handling of the walking directions, then rename that to `transit_leg`.

Reply:
That's what I meant by this comment. I don't think it is in the scope of this ticket to refactor the previous work. But, we should do it because the two different legs have completely different rendering methods. And, they should both take a `leg` instead of that destructuring.

Reply:
Yeah, I agree. I just think since this change makes part of the existing `leg/1` logic [unreachable](https://github.com/mbta/dotcom/blob/929128f3dbd6f6411f792b3bab4cee846d16e72f/lib/dotcom_web/components/trip_planner/leg.ex#L55-L58) already, it feels a little odd not to make that refactor here. Just my 2c.